cache/clock_cache.cc

```diff
@@ -530,6 +550,20 @@ bool ClockCacheShard::Release(Cache::Handle* handle, bool erase_if_last_ref) {
   }
 
   ClockHandle* h = reinterpret_cast<ClockHandle*>(handle);
+
+  if (h->IsDangling()) {
```

Comment:
Nice observation, like other immutable fields. And in normal operation, it's a very predictable branch.

Nit: without historical information, IIRC default branch prediction usually goes *into* an `if` rather than skipping over it (because on average, jumps are quite in the negative direction). We have a hint mechanism for cases like this: `if (UNLIKELY(...))`. Feel free to use LIKELY and UNLIKELY in more places when there's an "almost always" direction.